core/views.py

```diff
@@ -15,3 +25,34 @@ def profile(request):
         "max_exp": "2137",
     }
     return render(request, "core/profile.html", context={"user": user})
+
+
+def login(request):
+    return render(request, "core/login.html")
+
+
+def discord_login(_):
+    query = urlencode(
+        {
+            "client_id": CLIENT_ID,
+            "response_type": "code",
+            "redirect_uri": REDIRECT_URI,
+            "scope": "identify guilds.members.read",
+            "prompt": "none",
+        }
+    )
+    return redirect(f"https://discord.com/oauth2/authorize?{query}")
+
+
+def discord_code(request):
+    code = request.GET["code"]
+    auth_data = code_to_token(code)
```

Comment:
In case the code is not provided or the provided code is wrong, don't return people with 400/500 error. Instead, redirect to the login page